consul/src/main/scala/io/buoyant/consul/v1/ConsistencyMode.scala

```diff
@@ -0,0 +1,37 @@
+package io.buoyant.consul.v1
+
+import com.fasterxml.jackson.core.{JsonGenerator, JsonParser}
+import com.fasterxml.jackson.databind.{DeserializationContext, SerializerProvider}
+import io.buoyant.config.{ConfigDeserializer, ConfigSerializer}
+
+sealed trait ConsistencyMode
+
+object ConsistencyMode {
+
+  case object Default extends ConsistencyMode
+  case object Stale extends ConsistencyMode
+  case object Consistent extends ConsistencyMode
+
+}
+
+class ConsistencyModeDeserializer extends ConfigDeserializer[ConsistencyMode] {
+
+  override def deserialize(jp: JsonParser, ctxt: DeserializationContext): ConsistencyMode =
+    catchMappingException(ctxt) {
+      _parseString(jp, ctxt) match {
+        case "default" => ConsistencyMode.Default
+        case "stale" => ConsistencyMode.Stale
+        case "consistent" => ConsistencyMode.Consistent
+        case unknown =>
+          throw new IllegalArgumentException(s"Illegal Consul consistency mode level: $unknown")
+      }
+    }
+}
+
+class ConsistencyModeSerializer extends ConfigSerializer[ConsistencyMode] {
+  override def serialize(
+    mode: ConsistencyMode,
+    jgen: JsonGenerator,
+    provider: SerializerProvider
+  ): Unit = jgen.writeString(mode.toString.toLowerCase) // is it OK ro rely on `toString`...?
```

Comment:
Great, thanks! I'll update the comment with explanation (about good `toString` then). Also as we rely on `toString` to operate in some way - maybe it worth a test?..